packages/embark/src/lib/modules/ethereum-blockchain-client/index.js

```diff
@@ -20,15 +27,32 @@ class EthereumBlockchainClient {
     this.embark.registerActionForEvent('deployment:contract:beforeDeploy', this.determineAccounts.bind(this));
     this.events.request("blockchain:client:register", "ethereum", this.getClient.bind(this));
     this.events.request("deployment:deployer:register", "ethereum", this.deployer.bind(this));
+
+    this.registerAPIRequests();
+  }
+
+  get web3() {
+    return (async () => {
+      if (!this._web3) {
+        const provider = await this.events.request2("blockchain:client:provider", "ethereum");
+        this._web3 = new Web3(provider);
+      }
+      return this._web3;
+    })();
+  }
```

Comment:
We've been repeating this pattern over and over again. We should put that in some sort of service or something (blockchainConnector anyone?)